test/Renci.SshNet.Tests/Classes/Security/Cryptography/Ciphers/AesCipherTest.Gen.cs.txt

```diff
@@ -21,109 +21,130 @@ foreach ((string mode, (string modeCode, CipherMode? bclMode)) in modes)
 {
     foreach (int keySize in new int[] { 128, 192, 256 })
     {
-        foreach (int inputLength in new int[] { 16, 32, 64 })
+        foreach (int inputLength in new int[] { 16, 35, 64 })
         {
-            byte[] input = new byte[inputLength];
-            random.NextBytes(input);
+            foreach (bool pad in new bool[] { false, true })
+            {
+                // It is not allowed to use no padding on non-block lengths
+                // It makes sense in cfb, ctr and ofb modes
+                if (!pad && inputLength % 16 != 0 && mode is not "cfb" or "ctr" or "ofb")
```

Comment:
Yes, it is a little bit interesting. The word `or` here is for and **only for** `mode` which is not the same with `||` operator. With additional brackets, it might be easier for people who codes; without additional brackets, it might be easier for people who does not code.

Reply:
they have different behaviour due to precedence of the `not`:

```c#
int x = 3;

Console.WriteLine(x is not 1 or 2 or 3); // True
Console.WriteLine(x is not (1 or 2 or 3)); // False
```

It is not outputting e.g. `AES_CTR_128_Length35_NoPad` which I assume is not intended

Reply:
Ah I see. Thanks a lot!

Reply:
will merge once the tests are regenerated